### PR TITLE
[ITA-63] Added startedByTimer helper.

### PR DESCRIPTION
**Pull request by michal-raska**

### CODE REVIEW

**Review by jakubhava — DISMISSED** [review]

File: vars/startedByTimer.groovy [hunk 1]
```diff
@@ -0,0 +1,4 @@
+def call() {
+  // may fail if the sandbox permissions are missing
+  return currentBuild.rawBuild.getAction(CauseAction.class).findCause(hudson.triggers.TimerTrigger.TimerTriggerCause.class) != null
```
Comment (mmalohlava) [inline vars/startedByTimer.groovy]: Should it be here or in Utils groovy class?

File: vars/isPRJob.groovy [hunk 2]
```diff
@@ -0,0 +1,4 @@
+def call() {
+  // Only PR builds have the CHANGE_BRANCH set
+  return env.CHANGE_BRANCH != null && env.CHANGE_BRANCH != ''
```
Comment (mmalohlava) [inline vars/isPRJob.groovy]: i would attach the method to BuildInfo and store it as part of BuildInfo

File: vars/startedByTimer.groovy (hunk 1 above)
Comment (mmalohlava) [reply vars/startedByTimer.groovy]: moreover, store it as part of BuildInfo as well.

**Comment by michal-raska** [comment]
@mmalohlava I moved both helpers into the BuildInfo class

File: src/ai/h2o/ci/BuildInfo.groovy [hunk 3]
```diff
@@ -18,19 +18,27 @@ class BuildInfo implements Serializable {
 
     private String version
 
+    final private boolean prJob
+
+    final private boolean triggeredByTimer
```
Comment (mmalohlava) [inline src/ai/h2o/ci/BuildInfo.groovy]: This is good idea, but it would be better to have triggerReason as enum (e.g., {manual, timer, external})

**Review by mmalohlava — COMMENTED** [review]
I would say we should provide all reasons how job can be triggered, not only timer..

**Comment by mmalohlava** [comment]
@michal-raska ^ there is a conflict now. Let's fix it and merge it.

**Pull request closed without merging**